src/components/Main/MainView/MessageInput/use/wordCompleter.ts

```diff
@@ -0,0 +1,115 @@
+import store from '@/store'
+import createTree from '@/lib/trieTree'
+import { nextTick, ComputedRef, WritableComputedRef, Ref } from 'vue'
+import textFieldMirror from './textFieldMirror'
+import { animeEffectSet, sizeEffectSet } from '@traptitech/traq-markdown-it'
+
+const getCurrentWord = (elm: HTMLTextAreaElement, text: string) => {
+  text = text.replaceAll('　', ' ')
+  const startIndex = elm.selectionStart
+  const prevAtMarkIndex = text.lastIndexOf('@', startIndex - 1)
+  const prevColonIndex = text.lastIndexOf(':', startIndex - 1)
+  const prevPeriodIndex = text.lastIndexOf('.', startIndex - 1)
+  const nearest = Math.max(prevAtMarkIndex, prevColonIndex, prevPeriodIndex)
+  const begin = nearest < 0 ? 0 : nearest
+  const end = elm.selectionEnd
+  const word = text.substring(begin, end)
+  const prevSpaceIndex = text.lastIndexOf(' ', startIndex - 1)
+  const divided = prevSpaceIndex > nearest
+  return { word, begin, end, divided }
+}
+
+const getDeterminedCharacters = (candidates: string[]) => {
+  const minLength = Math.min(...candidates.map(c => c.length))
+  const determined: string[] = []
+  for (let i = 0; i < minLength; i++) {
+    determined[i] = [...candidates[0]][i]
+    for (const candidate of candidates) {
+      if (determined[i] !== [...candidate][i]) {
+        return determined.slice(0, determined.length - 1).join('')
+      }
+    }
+  }
+  return determined.join('')
+}
+
+const useWordCompleter = (
+  textareaRef: ComputedRef<HTMLTextAreaElement | undefined>,
+  value: WritableComputedRef<string>,
+  hideSuggester: Ref<boolean>,
+  position: Ref<{ top: number; left: number }>,
+  suggesteCandidates: Ref<string[]>
```

Comment:
```suggestion
  suggestedCandidates: Ref<string[]>
```
これも`useWordCompleter`内で宣言して返すのがきれいだと思う